client/src/indexer/mod.rs

```diff
@@ -0,0 +1,131 @@
+use crate::watcher::WatcherListener;
+use anyhow::Result;
+use async_trait::async_trait;
+use grpc::file_manager_service_client::FileManagerServiceClient;
+use grpc::{File, FileEventType, FileRequest, FileResponse};
+use log::{debug, error, info, warn};
+use notify::DebouncedEvent;
+use std::ffi::OsStr;
+use std::path::Path;
+use tonic::transport::Channel;
+
+mod grpc;
+
+/// The `Indexer` is responsible to handle events on files
+/// and to synchronize those files onto the server.
+#[derive(Clone)]
+pub struct Indexer {
+    /// The file manager gRPC client
+    client: FileManagerServiceClient<Channel>,
+}
+
+impl Indexer {
+    /// Bootstrap the server
+    pub async fn bootstrap() -> Result<Self> {
+        info!("starting server");
+
+        let client = FileManagerServiceClient::connect("http://localhost:8090").await?;
```

Comment:
Can you create a simple config builder so specify by env var this ?

Reply:
refactored into #30, but another PR is coming to handle configuration in a better way